src/client/components/pages/parts/add-to-collection-modal.js

```diff
@@ -0,0 +1,309 @@
+import * as bootstrap from 'react-bootstrap';
+import CustomInput from '../../../input';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import ReactSelect from 'react-select';
+import SelectWrapper from '../../input/select-wrapper';
+import request from 'superagent';
+
+
+const {Alert, Col, Button, Modal} = bootstrap;
+
+class AddToCollectionModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			collectionsAvailable: [],
+			message: {
+				text: null,
+				type: null
+			},
+			selectedCollections: [],
+			show: this.props.show,
+			showCollectionForm: false
+		};
+
+		this.handleClose = this.handleClose.bind(this);
+		this.getCollections = this.getCollections.bind(this);
+		this.handleAddToCollection = this.handleAddToCollection.bind(this);
+		this.toggleRow = this.toggleRow.bind(this);
+		this.handleShowCollectionForm = this.handleShowCollectionForm.bind(this);
+		this.handleShowAllCollections = this.handleShowAllCollections.bind(this);
+		this.handleAddToNewCollection = this.handleAddToNewCollection.bind(this);
+		this.isValid = this.isValid.bind(this);
+	}
+
+	async componentDidUpdate(prevProps) {
+		const collections = await this.getCollections();
+		if (prevProps.show !== this.props.show) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({collectionsAvailable: collections, show: this.props.show});
+		}
+	}
+
+	toggleRow(collectionId) {
+		// eslint-disable-next-line react/no-access-state-in-setstate
```

Comment:
That does not look good. Why is this needed?

Reply:
Yeah this is not needed. I forgot to remove it